app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java

```diff
@@ -58,21 +60,33 @@ public void onCreate(Bundle savedInstanceState) {
 
         mLocationFetcher = LocationFetcher.getInstance(this);
 
-        try {
-            tryAddProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_GPS);
-            tryAddProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_NETWORK);
-            tryAddProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_PASSIVE);
-
-            mLocationFetcher.requestLocationUpdates(mLocationMinUpdatesMs, mLocationMinDistanceM);
+        tryAddProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_GPS);
+        tryAddProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_NETWORK);
+        tryAddProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_PASSIVE);
 
-            //if the registration was OK and no SecurityException was thrown
-            registerLocationReceiver();
-            startMainActivity();
-        } catch (SecurityException e) {
+        // Request for log permissions before we use it
+        if (!doesHaveGpsPermissions()) {
             LocationFetcher.askForLocationPermission(this);
+        } else {
+            requestGpsLocationUpdates();
+            startMainActivity();
         }
     }
 
+    private void requestGpsLocationUpdates() {
+        if (!mLocationFetcher.getIsRequestingUpdates()) {
+            mLocationFetcher.requestLocationUpdates(mLocationMinUpdatesMs, mLocationMinDistanceM);
```

Comment:
This line can throw an exception if it doesn't have any provider attached to it. (in-case all the tryAddProvider method calls failed).
try-catch that and log appropriately. !fix